### DISCUSSION TITLE
Detect Unknown Libre 2 SN# sensors as Libre 2

### DISCUSSION THREAD

**Discussion by parapenT1sta** [NightscoutFoundation/xDrip#2845] [Ideas]
Following this topic: https://github.com/NightscoutFoundation/xDrip/discussions/2840.

Do you think is a good idea to make all the new unknow serial numbers for Libre 2 detect the sensor as Libre 2 instead as Libre 1 as it is right now? Since it's harder and harder to find Libre 1 in our days and most of the Libre 1 path info are already well known (I Think).

So my suggestion is to return Libre2 as failsafe.

![image](https://github.com/NightscoutFoundation/xDrip/assets/608650/b723d4b3-ef99-4f13-82a5-49c7a595780c)

**Comment by tzachi-dar** [comment]
I think you have a point here, but I want to make sure that we are not missing anything here.
So right now the fix is only for the SN that we know.
If some one can verify that this two works this would be great.

For replacing the xdrip version one will have to create a backup. remove xdrip from the phone, install the new version and reload the backup.

New version of xdrip is at: https://drive.google.com/file/d/1_wZ1oTWuKcq-lThl5RWGL9Vi-ajmqFGk/view?usp=sharing
New version of oop2 is at: https://drive.google.com/file/d/1f1VHW2I8w7Xe3kSQqdaY3kihPLs47ILS/view?usp=sharing

**Reply by idalioreis** [reply]
It work, thank 
It can be necessary turn off Google play protect for install xdrip.👍👍👍

**Reply by jaky1977** [reply]
Thanks you, it's working!

**Reply by tjcbento** [reply]
Can confirm this is working for me as well. Where do I send the celebration beers? 😀

**Reply by tjcbento** [reply]
~~Might have spoken a bit too soon. NFC reads works flawlessly. Bluetooth is not working (even with the suggested version of OOP2).~~
Just realised I am dumb. Disregard please 😄

**Reply by matildelima** [reply]
Good afternoon.  In my daughter's case, it's still not working.  I uninstalled both apps and installed the ones you indicated.  xdrip still cannot find the sensor.

**Reply by parapenT1sta** [reply]
> Good afternoon. In my daughter's case, it's still not working. I uninstalled both apps and installed the ones you indicated. xdrip still cannot find the sensor.

Share the information requested on this topic: https://github.com/NightscoutFoundation/xDrip/discussions/2840

It has been working for at least 5 people.

**Reply by LTorrao** [reply]
working here :) thank you

**Reply by sebmatrix86** [reply]
I also had the same problem. With this new version, my sensor is now detected. Thank you so much !

**Reply by jo-wur** [reply]
Same problem, I can confirm that this works for me as well. I am very happy that I can use xdrip again with my Libre 2. Thank you!

**Comment by jim-ip** [comment]
Looks like this one has gone quiet, so I'm guessing that the fix has been finalised. Just in case it's any help, I just got my first batch of these new sensors, and the fix from @tzachi-dar works perfectly - *huge* thanks.

I don't know if you've gone with specific serial numbers or  have defaulted to Libre 2, so I've attached a screenshot with the patchInfo for my sensor just in case it's of any use.

FWIW, my *original* Libre1 reader works fine with the new sensors.


![Screenshot_2023-07-06-21-04-30-08_2dd2c1bcb8f659cd696bcba2072f751b](https://github.com/NightscoutFoundation/xDrip/assets/3473047/3b5a581d-90bb-40a4-be18-8d1f69662afc)

**Comment by DayZer035** [comment]
This worked for me when I updated both apps to recommended versions, Thank you!

**Comment by pkviet** [comment]
Hello,
@tzachi-dar 
it seems the build for xdrip+ requires permission. I can't download it. I've been affected by this bug for the past three months. It'd be nice to PR and merge the fix + push a new release.
Thanks a lot.

**Reply by pkviet** [reply]
seems there's already a PR which was merged.
https://github.com/NightscoutFoundation/xDrip/pull/2847

**Reply by pkviet** [reply]
tested the PR through nightly and my Libre 2 EU issues are now gone.
Thanks for the fix @tzachi-dar

**Comment by ollieglass** [comment]
Hi, excuse me re-opening this but I'm getting the same issue. I have the latest OOP (version `27da3f5-2023.05.25`) and XDrip+ (version `23ccbd1-2023.07.16`).

I'm using an EU Libre 2. LibreView can scan it, XDrip+ gives an NFC read timeout.

I'm using a Pixel 3a with Android 12. I've restarted the phone a few times. Is there anything else I could try?

![Screenshot_20230924-124651](https://github.com/NightscoutFoundation/xDrip/assets/311098/590ae6f9-de4c-46f8-8468-26024c6d1e4f)

**Reply by pkviet** [reply]
Mmh i have the very same versions of xdrip and oopalgo2.
It works fine with my pixel 7 pro + pixel watch. Both LibreLink and xdrip+ work fine concurrently.
Did you start the EU patch with LibreLink?
Did you make sure to force stop it for one hour before NFC scanning it with xdrip+ ?
If you did, I wonder if it could be a case of new serials...

**Reply by ollieglass** [reply]
Hi @pkviet 

Thanks for the info and ideas. However I didn't do any of these things. 

I just read up about the EU patch for LibreLink. I've not used it before. I've never force stopped it either. Usually I scan, uninstall, wait for a while, then scan with XDrip+

I've been using XDrip+, OOP and EU Libre 2 sensors for about a year with no issues. Only my last sensor and my current one have had this issue. 

I updated to the latest version, still had this issue, restarted a few times, still has the issue, then I posted here.

Is there a way I can share the serial with you or check it against a list somehow?

**Comment by pkviet** [comment]
@ollieglass 

> 
> I just read up about the EU patch for LibreLink. I've not used it before. I've never force stopped it either. Usually I scan, uninstall, wait for a while, then scan with XDrip+

I didn't mean the "patched LibreLink", but meant the Freestyle Libre 2 patch.
What I do is:

-  start it on LibreLink;
- force stop LibreLink;
- NFC scan on xdrip+ after one hour, 
- wait for calibration after three glucose values have been retrieved,
- reopen LibreLink and scan

> 
> I've been using XDrip+, OOP and EU Libre 2 sensors for about a year with no issues. Only my last sensor and my current one have had this issue.
> 
> I updated to the latest version, still had this issue, restarted a few times, still has the issue, then I posted here.
> 
> Is there a way I can share the serial with you or check it against a list somehow?

go to setting->more settings ->extra tags for logging and add there NFCReaderX:v
After that please scan again, you should see in the log something like:
'' patchInfo = 0xdddddd"
Report the number here.
The working numbers are listed here for Libre 2 : 
 0x9d0830
 0xc50930
(source in code is 
https://github.com/NightscoutFoundation/xDrip/blob/10f2319be05be5a1f73ad87ff3a2b2c900272818/app/src/main/java/com/eveningoutpost/dexdrip/models/LibreOOPAlgorithm.java#L304-L305 )
If you have a different number, it'll require to be added in the code. I can open a Pull Request for that if needs be.

**Reply by ollieglass** [reply]
Thank you. I'm so sorry but I realise I'm mistakenly using a Libre 1 instead of a Libre 2. 

I appreciate your time and help, and I'm sorry for the unnecessary error report.

**Comment by pkviet** [comment]
> Thank you. I'm so sorry but I realise I'm mistakenly using a Libre 1 instead of a Libre 2.
> 
> I appreciate your time and help, and I'm sorry for the unnecessary error report.

even if it's a Libre 1, it should not error out. Can you post the patchinfo hex as explained above ?

**Reply by ollieglass** [reply]
Sure, here's the log when I scan it

![Screenshot_20230924-170704](https://github.com/NightscoutFoundation/xDrip/assets/311098/a492c398-0bfd-400e-8421-c8432f449597)

**Reply by pkviet** [reply]
your patchinfo hex is right for a Libre 1 new patch ... Means your issue is different. Can't help beyond that, sorry !

**Comment by marinko1973** [comment]


**Reply by Navid200** [reply]
What you are describing is the difference in behavior between two phones.

This thread is about something else, which is not unique to any particular phone.
It's your choice what you want to do.  But, I strongly advise against using this thread.

**Reply by marinko1973** [reply]
her phone was running fine last six months and have changed behaviour after we replaced new sensor.  I am ready to move my question elsewhere. Any suggestion?

**Reply by parapenT1sta** [reply]
Update xDrip and mainly OOP2.

**Reply by marinko1973** [reply]
My original message has been deleted here and transferred to discussion since the topic of my message was different from this thread. Thanks

**Comment by muccioli** [comment]
Good morning. I don't know if this is directly related to the subject, but I have the following problem:
NFC READ TIME OUT
I installed the nescessari OOP2 and xdrip+ applications, I believe in their most recent versions. I followed all the settings, but the error continues. In the Abbott app, LibreLink works perfectly. The detail is that I live in Brazil and the latest version says it is compatible with Libre 2 US and EU. There is a way to read and start the sensor in xdrip+?
![WhatsApp Image 2024-06-13 at 17 48 48](https://github.com/NightscoutFoundation/xDrip/assets/118318032/c46c3e87-2acf-45ed-86b1-db60526b63a7)

**Reply by parapenT1sta** [reply]
Install the latest OOP2 available here: https://github.com/NightscoutFoundation/xDrip/discussions/3439#discussioncomment-9611227

And this version Xdrip+ version: https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.06.11

**Reply by muccioli** [reply]
Thanks. I trie it.

**Reply by muccioli** [reply]
Not works. Same error: NFC read timeout

**Reply by PhilTU** [reply]
Can you confirm versions of xDrip+ and OOP2 by number please

Nightly release https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.06.11 includes code for UK L2+ sensor.

If you add tags to your logging to get the patchinfo of the sensor it may shed more light...

go to setting->more settings ->extra tags for logging and add there NFCReaderX:v
After that please scan again, you should see in the log something like:
'' patchInfo = 0xdddddd"

**Reply by muccioli** [reply]
Thank you very much.
I've already managed to make the connection.

**Reply by talk2lurch** [reply]
> Thank you very much. I've already managed to make the connection.

May I ask you how you solved the problem? I have exactly the same situation as yours and still no good result

**Reply by talk2lurch** [reply]
> Can you confirm versions of xDrip+ and OOP2 by number please
> 
> Nightly release https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.06.11 includes code for UK L2+ sensor.
> 
> If you add tags to your logging to get the patchinfo of the sensor it may shed more light...
> 
> go to setting->more settings ->extra tags for logging and add there NFCReaderX:v After that please scan again, you should see in the log something like: '' patchInfo = 0xdddddd"

I have exactly the same situation as @muccioli. I followed all the instructions here and the problem persists. It's my first Libre2 sensor. 

My xDrip+ version is **1958827-2024.06.11** and OOP2 is **27da3f5-2024.05.31**.

Any alternative seen? Abbott has just released the Libre2+ UK in the Brazilian market, wouldn't it be related to unknown serial numbers?

![Screenshot_20240630-144503](https://github.com/NightscoutFoundation/xDrip/assets/80027732/5fcfdb0e-18de-4ff2-84f6-01834ace8d0c)

**Reply by parapenT1sta** [reply]
Your PatchID is not in xDrip source code. It might not be supported by OOP2 yet. Maybe @tzachi-dar can confirm about it.

```
            case 0xdf0000:
                return SensorType.Libre1;
            case 0xa20800:
                return SensorType.Libre1New;
            case 0xe50003:
            case 0xe60003:
                return SensorType.LibreUS14Day;
            case 0x9d0830:
            case 0xc50930:
            case 0xc60931:
                return SensorType.Libre2;
            case 0x700010:
                return SensorType.LibreProH;
```

**Comment by gui-dos** [comment]


**Reply by parapenT1sta** [reply]
@tzachi-dar

**Comment by mhalano** [comment]
Any solutions? I have one of these Libre 2 Plus since I live in Brazil. I got NFC read timeout and the log appears this part talking about identifying as libre1 as failsafe.

**Reply by PhilTU** [reply]
Brazil sensors not currently supported I think

**Comment by WHSchellekens** [comment]
I had the same issues of my Libre 2 sensor not being recognized, with the Libre 1 fallback. I updated my xDrip version to a nightly build with [these instrcutions](https://navid200.github.io/xDrip/docs/Updates.html), then it worked again! I assume the version/serials of the Libre sensors I received were not yet supported by the stable xDrip build.

**Reply by PhilTU** [reply]
Yes, the stable release was 2 weeks before a new sensor type was released, this happened twice last year so we were always on the back foot.
